src/azure-cli/azure/cli/command_modules/storage/operations/fileshare.py

```diff
@@ -93,3 +93,57 @@ def get_share_stats(client, timeout=None, **kwargs):
 def set_share_metadata(client, metadata=None, timeout=None, **kwargs):
     client.set_share_metadata(metadata=metadata, timeout=timeout, **kwargs)
     return True
+
+
+def list_handle(client, marker, num_results, **kwargs):
+    from ..track2_util import list_generator
+    directory_path = kwargs.pop("directory_name", None)
+    if directory_path and directory_path.startswith('./'):
+        directory_path = directory_path.replace('./', '', 1)
+    dir_client = client.get_directory_client(directory_path=directory_path)
+    file_name = kwargs.pop("file_name", None)
+    if file_name:
+        total_path = directory_path + '/' + file_name if directory_path else file_name
+        dir_client = client.get_directory_client(directory_path=total_path)
+        if not dir_client.exists():
+            dir_client = client.get_directory_client(directory_path=directory_path)
+            client = dir_client.get_file_client(file_name=file_name)
+            kwargs.pop("recursive")
+        else:
+            client = dir_client
+    else:
+        client = dir_client
+
+    generator = client.list_handles(results_per_page=num_results, **kwargs)
+    pages = generator.by_page(continuation_token=marker)  # SharePropertiesPaged
+    result = list_generator(pages=pages, num_results=num_results)
+
+    if pages.continuation_token:
+        next_marker = {"nextMarker": pages.continuation_token}
+        result.append(next_marker)
+
+    return result
+
+
+def close_handle(client, **kwargs):
+    directory_path = kwargs.pop("directory_name", None)
+    if directory_path and directory_path.startswith('./'):
+        directory_path = directory_path.replace('./', '', 1)
+    dir_client = client.get_directory_client(directory_path=directory_path)
+    file_name = kwargs.pop("file_name", None)
+    if file_name:
+        total_path = directory_path + '/' + file_name if directory_path else file_name
+        dir_client = client.get_directory_client(directory_path=total_path)
+        if not dir_client.exists():
+            dir_client = client.get_directory_client(directory_path=directory_path)
+            client = dir_client.get_file_client(file_name=file_name)
+            kwargs.pop("recursive")
+        else:
+            client = dir_client
+    else:
+        client = dir_client
```

Comment:
Seems to be the same with `list_handle()`. Can we extract a common function for this?